Add relationship support and full community mapper list

kspurgin opened the pull request:
Adds support for authorityhierarchy, objecthierarchy, and nonhierarchicalrelationship imports

The processing step for nonhierarchicalrelationship creates two reciprocal records that must be transferred for each row of the input CSV. Support for this introduces the following changes: 

- addition of "row_occurrence" for use in identifying info/status, warnings, and errors for report compilation and cached mapping results
-  transfer step now uses the full processing report as its input, rather than the original input file, so that it has the full list of records to transfer

Minor changes to seeds and tests to deal with [newly specified profile naming conventions](https://github.com/collectionspace/cspace-config-untangler#config-and-resulting-mappertemplate-naming-conventions) (to support determining profile name/version unambiguously) and new full mapper manifest.

Change `Mapper` title format to support selecting mapper by typing in batch creation form

Minor change to stdout while a process job is running, to make it more informative

mark-cooper reviewed (changes requested):
Hey Kristina,

Looks good. Don't worry about this right now but I'm going to add Rubocop style checks soon to enforce consistency with layout / formatting. Passing those checks will be required to have an approved PR (most code editors have a Rubocop plugin available).

There's also a broken test to do with the update for mappers, but that's an easy fix I'll do after merging.

Couple of things I'd like to see:

- Only one class per file. RecordActionChecker (and friends) should be separate.
- Service classes are usually pretty easy to test and should have them. If you just want to add placeholder files for now that's ok (tests/services), we can get to them later.
- You know this better than I do at this point -- delete any services (and things in general) that are obviously not being used (like TermManagerService and corresponding tests).

That's it 👍

kspurgin commented on config/environments/test.rb:
```diff
@@ -22,8 +22,8 @@
   config.consider_all_requests_local       = true
-  config.action_controller.perform_caching = false
-  config.cache_store = :null_store
+  config.action_controller.perform_caching = true
```
@mark-cooper -- was not sure how to test a service class like `RecordCacheService` with the existing settings here.

kspurgin replied:
Figured this out...

kspurgin commented on test/services/step_manager_service_test.rb:
```diff
@@ -4,21 +4,37 @@
 
 class StepManagerServiceTest < ActiveSupport::TestCase
```
@mark-cooper I added some tests here for public methods that I had added (and some behavior I changed). But a fair amount of the mess I added in here manipulates CSVs behind the scenes to generate the final report for each step without storing individual CSV row data in the database or caching it. Testing this stuff requires actual data in an input CSV associated with a batch, and the ability to verify the resulting CSV(s) are as expected in the tests. The fixtures as currently set up don't seem to support testing this and I was unable to figure out how to fix that today.

kspurgin commented:
@mark-cooper - ok, I have done the following: 
- moved classes into their own files (think I meant to do that after verifying that they worked, and forgot)
- written tests for services where I could figure out how; made placeholders otherwise
- removed unused services
- various other cleanup/improvement noticed while working on the above (decoupling things to ease testing, etc.)
